test/csuite/wt9199_checkpoint_txn_commit_race/main.c

```diff
@@ -0,0 +1,274 @@
+  "create,cache_size=2GB,statistics=(all),statistics_log=(json,on_close,wait=1),,timing_stress_for_"
+  "test=[commit_transaction_slow]";
+static const char table_config_row[] =
+  "allocation_size=4KB,leaf_page_max=4KB,key_format=Q,value_format=Q";
+static const char *const uri = "table:wt9199-checkpoint-txn-commit-race";
+static uint64_t global_stable_ts;
+
+static bool inserted;
+
+/* Structures definition. */
+struct thread_data {
+    WT_CONNECTION *conn;
+    const char *uri;
+};
+
+/* Forward declarations. */
+static void run_test(const char *);
+static void *thread_func_checkpoint(void *);
+static void *thread_func_insert_txn(void *);
+void validate(void *);
+static void populate(WT_SESSION *);
+
+/*
+ * main --
+ *     Methods implementation.
+ */
+int
+main(int argc, char *argv[])
+{
+    TEST_OPTS *opts, _opts;
+
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+    testutil_check(testutil_parse_opts(argc, argv, opts));
+
+    printf("\n");
+    printf("Running test ...\n");
+    run_test(opts->home);
+
+    /* Cleanup */
+    if (!opts->preserve)
+        testutil_remove(opts->home);
+
+    return (EXIT_SUCCESS);
+}
+
+/*
+ * run_test --
+ *     Run test.
+ */
+static void
+run_test(const char *home)
+{
+    struct thread_data td;
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    pthread_t thread_checkpoint, thread_insert_txn;
+
+    inserted = false;
+    testutil_recreate_dir(home);
+    testutil_check(wiredtiger_open(home, NULL, conn_config, &conn));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create and populate table. Checkpoint the data after that. */
+    testutil_check(session->create(session, uri, table_config_row));
+
+    populate(session);
+    testutil_check(session->checkpoint(session, NULL));
+
+    td.conn = conn;
+    td.uri = uri;
+
+    /* Spawn checkpoint and insert threads. */
+    testutil_check(pthread_create(&thread_insert_txn, NULL, thread_func_insert_txn, &td));
+    testutil_check(pthread_create(&thread_checkpoint, NULL, thread_func_checkpoint, &td));
+
+    /* Wait for the threads to finish the work. */
+    (void)pthread_join(thread_insert_txn, NULL);
+    (void)pthread_join(thread_checkpoint, NULL);
+
+    validate(&td);
+
+    testutil_check(session->close(session, NULL));
+    session = NULL;
+
+    testutil_check(conn->close(conn, NULL));
+    conn = NULL;
+}
+
+/*
+ * thread_func_checkpoint --
+ *     Function to checkpoint the database.
+ */
+static void *
+thread_func_checkpoint(void *arg)
+{
+    struct thread_data *td;
+    WT_SESSION *session;
+    uint64_t stable_ts;
+    char ts_string[WT_TS_HEX_STRING_SIZE];
+    char tscfg[64];
+
+    td = (struct thread_data *)arg;
+
+    testutil_check(td->conn->open_session(td->conn, NULL, NULL, &session));
+
+    while (1) {
+        if (inserted) {
+            /* Wait for sometime to let the commit transaction checks the timestamp validity. */
+            __wt_sleep(5, 0);
```

Comment:
Does this need to be 5 seconds? I think 1 will work equally well since `__wt_timing_stress` is a relatively small delay window. I'll discuss this with you in person

Reply:
Fixed in 746beb4a74be524a787cd7d49ab66860f4b9617c